server/src/internalClusterTest/java/org/elasticsearch/index/shard/IndexShardIT.java

```diff
@@ -160,7 +160,10 @@ public void testDurableFlagHasEffect() {
             Translog.Location lastWriteLocation = tlog.getLastWriteLocation();
             try {
                 // the lastWriteLocaltion has a Integer.MAX_VALUE size so we have to create a new one
-                return tlog.ensureSynced(new Translog.Location(lastWriteLocation.generation, lastWriteLocation.translogLocation, 0));
+                return tlog.ensureSynced(
+                    new Translog.Location(lastWriteLocation.generation, lastWriteLocation.translogLocation, 0),
+                    SequenceNumbers.NO_OPS_PERFORMED
```

Comment:
I had to revert this suggested change because `SequenceNumber.max` will throw if you try to max compare with `UNASSIGNED_SEQ_NO`. The documentation on the method also calls out that this is excepted.

So that is why I reverted. Because any other change would be more significant. I guess my thought process is that any sync request is fine to request that the initial starting "state" be synced (which it always will). But let me know if you want something else here.

Reply:
I brought back this change by skipping the `SequenceNumber.max` check when it == `UNASSIGNED_SEQ_NO `.